open_samus_returns_rando/pickup.py

```diff
@@ -66,27 +50,110 @@ def patch_pickups(editor: PatcherEditor, pickups_config: list[dict]):
         set_custom_params["Param1"]["value"] = item_id
         set_custom_params["Param2"]["value"] = quantity
 
-        actordef_id = f"randomizer_powerup_{i}"
+        script["functions"][0]["params"]["Param1"]["value"] = \
+            'actors/items/randomizer_powerup/scripts/randomizer_powerup.lc'
+        script["functions"][0]["params"]["Param2"]["value"] = self.lua_editor.get_script_class(self.pickup,
+                                                                                               actordef_name=bmsad[
+                                                                                                   "name"])
+        return bmsad
+
+
+    def patch_multi_item_pickup(self, bmsad: dict) -> dict:
+        pickable: dict = bmsad["components"]["PICKABLE"]
+        script: dict = bmsad["components"]["SCRIPT"]
+
+        set_custom_params: dict = pickable["functions"][0]["params"]
+        # FIXME: I would set it to "ITEM_NONE" but the game is crashing if you pick up a second progressive
+        set_custom_params["Param1"]["value"] = "ITEM_WEAPON_WAVE_BEAM"
```

Comment:
This is sadly very ugly atm.

The game always opens the inventory / samus screen and show you what you collected with the text on the bottom. If I set it to "ITEM_NONE" it kinda saves a "I want to show you something" state.
The next time you pick another progressive up, it just opens that inventory screen again, adds the new item without highlighting it and just stucks there and doesn't take any input. So, you have to close emulator.

It even persists after save and reload but NOT after save, go to menu and load.

Maybe we could do an ugly hack like giving a missile expansion with an amount 0 or sth. Dunno. Can look at it in the next days.

Reply:
I might have a solution for this